internal/release/scripts/prep-release.sh

```diff
@@ -2,12 +2,27 @@
 
 # Helper script to prepare a release for the Go SDK.
 
-# Read the current build number from version-build to make sure the build number has been updated
-current_build_number=$(< internal/release/version-build)
+# Read current build and version for backup as well as comparison to new build
+current_build=$(< internal/release/version-build)
+current_version=$(< internal/release/version)
 
 version_file="internal/release/version"
 build_file="internal/release/version-build"
 
+# Function to execute upon exit
+cleanup() {
+    echo "Performing cleanup tasks..."
+    # Remove changelog file if it exists
+    rm -f "${changelog_file}"
+    # Revert changes to file if any
+    echo "${current_version}" > "${version_file}"
+    echo "${current_build}" > "${build_file}"
+    exit 1   
+}
+
+# Set the trap to call the cleanup function on exit
+trap cleanup ERR SIGINT
```

Comment:
Since the versions check happen inside a loop, and won't error out, the only signal we need to worry about is `SIGINT`
```suggestion
trap cleanup SIGINT
```

Reply:
Done